### PR TITLE
[SYCL][ESIMD][E2E] Add DG2 unified memory block_store tests

### PR DESCRIPTION
**Pull request by sarnex**
DG2 tests for block_store, I ran tests manually on PVC Linux and DG2 Linux/Windows

The SLM accessor tests fail everywhere on DG2 on both Linux and Windows, so I just disabled them directly because the runtime driver check ends up being complicated, and none of the drivers, not even bleeding edge, work. I will file an internal bug for this.

### CODE REVIEW

File: sycl/test-e2e/ESIMD/unified_memory_api/block_store_slm_acc_dg2.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+//==- block_store_slm_acc_dg2.cpp - DPC++ ESIMD on-device test -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//==------------------------------------------------------------==//
+// TODO: Enable after driver bug is fixed
+// XFAIL: *
```
[the remark this answers is not in the document]
Comment (sarnex) [reply sycl/test-e2e/ESIMD/unified_memory_api/block_store_slm_acc_dg2.cpp:9]: Every or almost every case fails, will send the tracker soon

[the remark this answers is not in the document]
Comment (v-klochkov) [reply sycl/test-e2e/ESIMD/unified_memory_api/block_store_slm_acc_dg2.cpp:9]: Ok, it is interesting that `slm` passes, and `slm_acc` fails.
Thank you!

**Review by v-klochkov — APPROVED** [review]

File: sycl/test-e2e/ESIMD/unified_memory_api/block_store_slm_acc.cpp [hunk 2]
```diff
@@ -5,6 +5,9 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //==----------------------------------------------------------==//
+// TODO: Enable after driver bug is fixed
+// XFAIL: gpu-intel-dg2
```
[the remark this answers is not in the document]
Comment (sarnex) [reply sycl/test-e2e/ESIMD/unified_memory_api/block_store_slm_acc.cpp:9]: Sorry, I was wrong. We do set it for postcommit [here](https://github.com/intel/llvm/blob/19cd6144aaed64719475c4f4a0ee626bdbc428dc/.github/workflows/sycl_post_commit.yml#L47). I think this PR is safe to merge.

**Comment by sarnex** [comment]
@intel/llvm-gatekeepers Can we merge this one? It The Arc precommit job won't run the new tests added here anyway, they will only run in postcommit and I am convinced it will pass based on local testing and checking the flags passed to lit in the postcommit CI.

**Comment by aelovikov-intel** [comment]
Arc runner is down, I've reported it.

**Pull request merged**